### PR TITLE
Add TrainStepInfo used on the new frontend API

### PR DESCRIPTION
**Pull request by thiagocrepaldi**
This is part of https://github.com/microsoft/onnxruntime/issues/4176

### CODE REVIEW

**Comment by ytaous** [comment]
Seems a lot of duplicates with the other PR.

**Comment by spandantiwari** [comment]
Looks good.

File: orttraining/orttraining/python/training/orttrainer.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
+from .optim import _OptimizerConfig
+
+class TrainStepInfo(object):
+    r"""Private class used to store runtime information from current train step.
+
+    After every train step, :py:meth:`ORTTrainer.train_step` updates the internal instance of
+    :py:class:`.TrainStepInfo` residing on :py:class:`.ORTTrainer` with relevant information
+    from the forward pass.
+
+    This class shouldn't be accessed directly by the user, unless they really know what they are doing.
```
Comment (spandantiwari) [inline orttraining/orttraining/python/training/orttrainer.py:10]: It is good to mention this. Maybe just add a few words: "This class is intended for use in ORTTrainer, and should not be accessed directly..."

Comment (thiagocrepaldi) [reply orttraining/orttraining/python/training/orttrainer.py:10]: The next sentence is something like ":py:class:`.ORTTrainer` passes it to relevant class methods automatically, such as :py:method:`._LRScheduler.get_lr` or :py:class:`.LossScaler.update`"

**Pull request merged**